### PR TITLE
Configure native dependencies from the makefile.

### PR DESCRIPTION
**Pull request by pchote**
Followup fixes to #7953.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -5,6 +5,11 @@ os=`uname`
 if [ "$os" == 'Linux' ]; then
 	locations=(/lib /lib64 /usr/lib /usr/lib64 /usr/lib/i386-linux-gnu /usr/lib/x86_64-linux-gnu /usr/local/lib /opt/lib)
 	sonames=(liblua.so.5.1.5 liblua5.1.so.5.1 liblua5.1.so.0 liblua.so.5.1 liblua-5.1.so liblua5.1.so)
+
+	if [ -f Eluant.dll.config ]; then
+		exit 0
```
Comment (Phrohdoh) [inline configure]: Success if no config?

File: configure [hunk 2]
```diff
@@ -22,9 +28,18 @@ if [ "$os" == 'Linux' ]; then
...
-	curl -Os https://raw.githubusercontent.com/OpenRA/OpenRALauncherOSX/master/dependencies/liblua.5.1.dylib
-	curl -Os https://raw.githubusercontent.com/OpenRA/OpenRALauncherOSX/master/dependencies/libSDL2.dylib
-	echo "Downloaded Lua 5.1 and SDL2 libraries successfully."
+	if [ ! -f Eluant.dll.config ]; then
+		echo "Fetching SDL2 config from GitHub."
+		curl -LOs https://raw.githubusercontent.com/OpenRA/OpenRALauncherOSX/master/dependencies/Eluant.dll.config
+	fi
+
+	if [ ! -f liblua.5.1.dylib ]; then
+		echo "Fetching OSX lua library from GitHub."
```
Comment (Phrohdoh) [inline configure]: `OS X` is preferred.

File: configure (hunk 1 above)
Comment (pchote) [inline configure]: Success if config.

File: configure (hunk 2 above)
Comment (pchote) [inline configure]: Good catch.

File: configure (hunk 1 above)
Comment (Phrohdoh) [inline configure]: Oh, of course. That's an embarrassing mistake.

**Comment by Phrohdoh** [comment]
:+1:

**Comment by pchote** [comment]
Fixed, and added a commit that moves `configure` &rarr; `thirdparty/configure-native-deps.sh` to address confusion/complaints on IRC about our nonstandard `configure` usage.  This command no longer needs to be run manually.

**Comment by Mailaender** [comment]
This is a very good idea. I saw distribution packagers getting confused and running our script with their usual %configure macros. Also it was already redundant to run, because the Makefile executed it, too for convenience.

There is just a problem that you made the script non-executable (chmod 0644 from 0755) which you will need to revert to not make this fail:

```
make native-dependencies
make: execvp: ./thirdparty/configure-native-deps.sh: Insufficient rights
Makefile:305: recipe for target 'native-dependencies' failed
make: *** [native-dependencies] Error 127
```

**Comment by pchote** [comment]
This appears to be a git bug.  The permissions are correct in my working copy, and discarding and recommiting the files (with verified good permissions) still shows 755 &rarr; 644 when pushed.

```
paul@localhost:thirdparty+more-dependency-fixes$ ls -lah *.sh
-rwxr-xr-x  1 paul  staff   1.5K  3 May 14:48 configure-native-deps.sh
-rwxr-xr-x  1 paul  staff   1.0K  3 May 14:47 fetch-thirdparty-deps-windows.sh
-rwxr-xr-x  1 paul  staff   2.5K  3 May 14:48 fetch-thirdparty-deps.sh
```

I'm not sure how to proceed from here.

**Comment by Mailaender** [comment]
I seem to be unaffected by that bug: see https://github.com/OpenRA/OpenRA/pull/8082.

**Pull request closed without merging**